frontend/packages/console-shared/src/hooks/useUserSettings.ts

```diff
@@ -99,43 +40,58 @@ export const useUserSettings = <T>(
   const [settings, setSettings] = React.useState<T>();
   const [loaded, setLoaded] = React.useState(false);
 
+  const [fallbackLocalStorage, setFallbackLocalStorage] = React.useState<boolean>(false);
+  const [lsData, setLsDataCallback] = useUserSettingsLocalStorage(
+    keyRef.current,
+    defaultValueRef.current,
+    fallbackLocalStorage,
+  );
+
   React.useEffect(() => {
     if (cfLoadError || (!cfData && cfLoaded)) {
       (async () => {
-        await getProject();
-        const cmCreated = await createConfigMap({
-          apiVersion: ConfigMapModel.apiVersion,
-          kind: ConfigMapModel.kind,
-          metadata: {
-            name: `user-settings-${userUid}`,
-            namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
-          },
-          data: {
-            ...(defaultValueRef.current !== undefined && {
-              [keyRef.current]: seralizeData(defaultValueRef.current),
-            }),
-          },
-        });
-        if (!cmCreated) {
-          setSettings(deseralizeData(defaultValueRef.current));
-          setLoaded(true);
+        // this would be replaced with proxy endpoint to create ConfigMap
+        try {
+          const projectExists = await getProject();
+          if (!projectExists) await createProject();
+          await createConfigMap({
+            apiVersion: ConfigMapModel.apiVersion,
+            kind: ConfigMapModel.kind,
+            metadata: {
+              name: `user-settings-${userUid}`,
+              namespace: USER_SETTING_CONFIGMAP_NAMESPACE,
+            },
+            data: {
+              ...(defaultValueRef.current !== undefined && {
+                [keyRef.current]: seralizeData(defaultValueRef.current),
+              }),
+            },
+          });
+        } catch (err) {
+          if (err?.response?.status === 403) {
+            setFallbackLocalStorage(true);
+          } else {
+            setSettings(defaultValueRef.current);
+            setLoaded(true);
+          }
```

Comment:
All other cases of an error are just ignored a.t.m? Can you add here a console.warn / error as well?

Reply:
we are logging for all in utils like updateConfigmap, createProject, get Project will add there for createConfigMap as well.